### DISCUSSION TITLE
Webpack.config.js error.

### DISCUSSION THREAD

**Discussion by dbiguenet** [iTwin/community#59] [Q&A]
Hi everyone,
Few days ago, I have updated iTwin/reality-capture dependencies because we have a lot of security issues.
Since then, some integration tests -using webpack- are not working anymore.

I get this error : 
`[webpack-cli] Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
│  - configuration.entry should be a non-empty array.
│    -> All modules are loaded upon startup. The last one is exported.`

I'm not sure why it doesn't work anymore since we are using the same version as the other repositories, such as itwinjs-core : 
```js
"webpack": "^5.94.0",
"webpack-cli": "^5.0.2"
```

Webpack.config.js : 
```js
const clientsLib = path.resolve(__dirname, "../../../lib/cjs");
const config = {
    mode: "development",
    entry: glob.sync(path.resolve(clientsLib, "test/**/*.test.js")),
    output: {
      path: path.resolve(clientsLib, "test/webpack/"),
      filename: "bundled-tests.js",
      devtoolModuleFilenameTemplate: "file:///[absolute-resource-path]"
    },
    devtool: "nosources-source-map",
   ...
```

I also tried to install an older version of webpack, still doesn't work.

Can you help me about this?

**Comment by aruniverse** [comment]
Please, link to your full webpack.config

**Comment by dbiguenet** [comment]
@aruniverse 
https://github.com/iTwin/reality-capture/blob/main/typescript/packages/reality-data-client/src/test/utils/webpack.config.js

**Comment by tcobbs-bentley** [comment]
Can you verify that `lib/cjs/test` includes `*.test.js` files at the time of the error? The error makes it seem like `glob.sync` returned an empty array.

**Reply by dbiguenet** [reply]
Yes, `lib/cjs/test` includes `*.test.js` files.
Looks like glob.sync doesn't work with windows paths. "resolve" method returns a windows path, so I have replaced "\\" with "/", and now it is working as expected. I'm not sure to understand though, we have not changed webpack.config.js file for months, and yet, the integration test worked...

**Reply by aruniverse** [reply]
I'm guessing your dependencies have changed. You are using `^` so you should verify the version installed is what you expect, and review the changelog for webpack. You can also pin you webpack dependencies to the last known good version/version that tests passed
